plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -578,12 +603,52 @@ public void testBrokerQueryWithTooManyRowsForSegmentQuery()
                 tooManyRowsTableValues);
     }
 
+    @Test
+    public void testMaxLimitForPassthroughQueries()
+            throws InterruptedException
+    {
+        assertQueryFails("SELECT string_col, updated_at_seconds" +
+                        "  FROM  \"SELECT updated_at_seconds, string_col FROM " + TOO_MANY_BROKER_ROWS_TABLE +
+                        "  LIMIT " + (MAX_ROWS_PER_SPLIT_FOR_BROKER_QUERIES + 1) + "\"",
+                "Broker query returned '13' rows, maximum allowed is '12' rows. with query \"select updated_at_seconds, string_col from too_many_broker_rows limit 13\"");
+
+        // Pinot issue preventing Integer.MAX_VALUE from being a limit: https://github.com/apache/incubator-pinot/issues/7110
+        assertQueryFails("SELECT * FROM \"SELECT string_col, long_col FROM " + ALL_TYPES_TABLE + " LIMIT " + Integer.MAX_VALUE + "\"",
+                "Unexpected response status: 500 for request \\{\"sql\" : \"select string_col, long_col from alltypes limit 2147483647\" \\} to url http://localhost:\\d+/query/sql, with headers \\{Accept=\\[application/json\\], Content-Type=\\[application/json\\]\\}, full response null");
+
+        // Pinot broker requests do not handle limits greater than Integer.MAX_VALUE
+        // Note that -2147483648 is due to an integer overflow in Pinot: https://github.com/apache/pinot/issues/7242
```

Comment:
Can we add validation on our end to disallow such queries? Or is this not easy to do due to dynamic tables?
Overflows can turn into unsigned values again if the value is sufficiently large (which is easy to do with typos).

Reply:
Yep, there is no easy way: the pinot request compiler will throw this error for dynamic tables when it parses the passthrough query.